src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_FIGURES_TYPES = 5;
+    private static final int SIZE_LIMIT = 5;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private static final String DEFAULT_COLOR = "White";
+
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_CIRCLE_RADIUS);
+    }
+
+    public Figure getRandomFigure() {
+        String randomColor = colorSupplier.getRandomColor();
+        int sideA = getRandomSize();
+        switch (random.nextInt(MAX_FIGURES_TYPES) + 1) {
+            case 1:
+                return new Circle(randomColor, sideA);

```

Comment:
It is not necessary to write return in all cases, it is possible to create a variable on method-level and populate it in cases

Reply:
Valid point. But as it's not said that we need to change figures after creation, but just draw them, I made classes immutable and it will cost me to rewrite half of the code to make it. Should I ? =)

Reply:
These statements are not connected, you can declare variable: Figure figure; And than populate it in cases:
figure = new Circle(randomColor, sideA); Where is the problem?

Reply:
Confuseв it with field setting, my bad